api/classes/publisher.js

```diff
@@ -1,411 +1,412 @@
 
-// SQL Query Generators
-var UserQueries = require('../modules/users/model').prototype.queries();
-var ProjectQueries = Projects.prototype.queries();
-var PublishedProjectQueries = PublishedProjects.prototype.queries();
-var PublishedFileQueries = require('../modules/publishedFiles/model').prototype.queries();
+const Projects = require(`../modules/projects/model`);
+const PublishedProjects = require(`../modules/publishedProjects/model`);
 
-var reject  = Promise.reject;
+// SQL Query Generators
+const usersQueryBuilder = require(`../modules/users/model`).prototype.queryBuilder();
+const projectsQueryBuilder = Projects.prototype.queryBuilder();
+const publishedProjectsQueryBuilder = PublishedProjects.prototype.queryBuilder();
+const publishedFilesQueryBuilder = require(`../modules/publishedFiles/model`).prototype.queryBuilder();
 
-/**
+/*
  * Utility functions
  */
 function success(type, username) {
   return function(message) {
-    log.info('Publish for ' + username + ' - [' + type + '] ' + message);
+    log.info(`Publish for ${username} - [${type}] ${message}`);
   };
 }
 
 function failure(type, username) {
-  return function(err) {
-    log.error({ error: err }, 'Publish for ' + username + ' - [' + type + ']');
-    reject(err);
+  return function(error) {
+    log.error({ error }, `Publish for ${username} - [${type}]`);
+    return Promise.reject(error);
   };
 }
 
-function getUploadRoot(client, user, project) {
+function getUploadRoot(publishClient, user, project) {
   if (!user || !project) {
     return null;
   }
 
-  var httpPrefix = '';
-  var httpClients = (process.env.HTTP_CLIENTS || '').split(',');
+  let httpPrefix = ``;
+  const httpClients = (process.env.HTTP_CLIENTS || ``).split(`,`);
 
   // If the project's publishing client matches to a list
   // of "can publish http://" clients, prefix the upload root
   // /with a special "HTTP" namespace.
-  if (client && httpClients.indexOf(client) !== -1) {
-    httpPrefix = '/HTTP';
+  if (publishClient && httpClients.indexOf(publishClient) !== -1) {
+    httpPrefix = `/HTTP`;
   }
 
-  return httpPrefix + '/' + user.name + '/' + project.id;
+  return `${httpPrefix}/${user.name}/${project.id}`;
 }
 
-function buildUrl(client, user, project) {
+function buildUrl(publishClient, user, project) {
   if (!user || !project) {
     return null;
   }
 
-  return process.env.PUBLIC_PROJECT_ENDPOINT + getUploadRoot(client, user, project);
+  return process.env.PUBLIC_PROJECT_ENDPOINT + getUploadRoot(publishClient, user, project);
 }
 
 // Takes an absolute path and uri-encodes each component
 // of the path to return a fully uri safe path
 function uriSafe(path) {
-  if (path === '/') {
-    return '/';
+  if (path === `/`) {
+    return `/`;
   }
 
-  var uriSafePath = '';
+  let uriSafePath = ``;
 
-  while (path !== '/') {
-    uriSafePath = Path.join('/', encodeURIComponent(Path.basename(path)), uriSafePath);
+  while(path !== `/`) {
+    uriSafePath = Path.join(
+      `/`,
```

Comment:
we're using the `\`/\`` string a lot here - let's catch it as a `const rootURL` or something?